api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -94,18 +95,22 @@
      * The configuration value is not guaranteed to be cached by the implementation, and may be expensive
      * to compute; therefore, if the returned value is intended to be frequently used, callers should consider storing
      * rather than recomputing it.
+     * <p>
+     * The result of this method is identical to the result of calling {@code getOptionalValue(propertyName, propertyType).get()}.
+     * In particular, if the value element of the given property name does not exist, a {@link NoSuchElementException} is thrown.
+     * This method never returns {@code null}.
      *
      * @param <T>
      *             The property type
      * @param propertyName
      *             The configuration property name
      * @param propertyType
      *             The type into which the resolved property value should get converted
-     * @return the resolved property value as an instance of the requested type
-     * @throws java.lang.IllegalArgumentException if the property cannot be converted to the specified type
-     * @throws java.util.NoSuchElementException if the property isn't present in the configuration
+     * @return the resolved property value as an instance of the requested type (not {@code null})
+     * @throws IllegalArgumentException if the property cannot be converted to the specified type
+     * @throws NoSuchElementException if the property does not exist
```

Comment:
NoSuchElementException if the property name or the value element of the given property name does not exist.